src/web/app/pages-instructor/instructor-course-edit-page/instructor-course-edit-page.component.ts

```diff
@@ -683,4 +692,146 @@ export class InstructorCourseEditPageComponent implements OnInit {
       this.statusMessageService.showErrorToast(resp.error.message);
     });
   }
+
+  /**
+   * Copies instructors from existing courses.
+   */
+  copyInstructors(): void {
+    this.isCopyingInstructor = true;
+    const courseTabModels: CourseTabModel[] = [];
+
+    forkJoin([
+      this.courseService.getAllCoursesAsInstructor('active'),
+      this.courseService.getAllCoursesAsInstructor('archived'),
+    ]).subscribe((values: any[]) => {
+      const activeCourses: Courses = values[0] as Courses;
+      const archivedCourses: Courses = values[1] as Courses;
+
+      activeCourses.courses.forEach((course: Course) => {
+        if (course.courseId !== this.courseId && course.institute === this.course.institute) {
+          const model: CourseTabModel = {
+            courseId: course.courseId,
+            courseName: course.courseName,
+            creationTimestamp: course.creationTimestamp,
+            isArchived: false,
+            instructorCandidates: [],
+            instructorCandidatesSortBy: SortBy.NONE,
+            instructorCandidatesSortOrder: SortOrder.ASC,
+            hasInstructorsLoaded: false,
+            isTabExpanded: false,
+            hasLoadingFailed: false,
+          };
+          courseTabModels.push(model);
+        }
+      });
+      archivedCourses.courses.forEach((course: Course) => {
+        if (course.courseId !== this.courseId && course.institute === this.course.institute) {
+          const model: CourseTabModel = {
+            courseId: course.courseId,
+            courseName: course.courseName,
+            creationTimestamp: course.creationTimestamp,
+            isArchived: true,
+            instructorCandidates: [],
+            instructorCandidatesSortBy: SortBy.NONE,
+            instructorCandidatesSortOrder: SortOrder.ASC,
+            hasInstructorsLoaded: false,
+            isTabExpanded: false,
+            hasLoadingFailed: false,
+          };
+          courseTabModels.push(model);
+        }
+      });
+    }, (err: ErrorMessageOutput) => {
+      this.isCopyingInstructor = false;
+      this.statusMessageService.showErrorToast(err.error.message);
+    }, () => {
+      const modalRef: NgbModalRef = this.ngbModal.open(CopyInstructorsFromOtherCoursesModalComponent);
+      modalRef.componentInstance.courses = courseTabModels;
+
+      modalRef.dismissed.subscribe(() => { this.isCopyingInstructor = false; });
+
+      modalRef.componentInstance.copyClickedEvent.subscribe((instructors: Instructor[]) => {
+        this.verifyInstructorsToCopy(instructors).subscribe((hasCheckPassed: boolean) => {
+          if (!hasCheckPassed) {
+            modalRef.componentInstance.isCopyingSelectedInstructors = false;
+            return;
+          }
+
+          this.addNewInstructors(instructors, modalRef);
+        });
+      });
+    });
```

Comment:
@wkurniawan07 Since you pointed out the `complete` block, can these be put inside the `next` block? I did some search but didn't find the execution order of `next` and `complete`. Can I assume for this specific case, `complete` block is executed after everything in the `next` block completes?